pages/api/profile/edit-profile.js

```diff
@@ -0,0 +1,80 @@
+import nextConnect from 'next-connect';
+import multer from 'multer';
+const AWS = require('aws-sdk');
+const { v4: uuid } = require('uuid');
+
+import User from '../../../models/User';
+import dbConnect from '../../../utils/dbConnect';
+
+const s3 = new AWS.S3({
+  accessKeyId: process.env.AWS_ID,
+  secretAccessKey: process.env.AWS_SECRET,
+});
+
+const storage = multer.memoryStorage();
+
+const router = nextConnect({
+  onError(error, req, res) {
+    res
+      .status(501)
+      .json({ error: `Sorry something Happened! ${error.message}` });
+  },
+  onNoMatch(req, res) {
+    res.status(405).json({ error: `Method '${req.method}' Not Allowed` });
+  },
+});
+
+router.use(multer({ storage }).single('image'));
+
+router.post(async (req, res) => {
+  const reqBody = JSON.parse(JSON.stringify(req.body));
```

Comment:
what is the structure of `req.body` here ? The json back and forth seems kinda redundant

Reply:
It's required since we're dealing with x-www-form data. There's no other way to parse it.

Reply:
limited other ways*** We need to user a bodyParser but I purposely have the disabled so it wouldn't conflict with multer.